test/Vanguard.test.ts

```diff
@@ -0,0 +1,212 @@
+    await stRIF.depositAndDelegate(deployer.address, votingPower).then(tx => tx.wait())
+    // give voting power to voter
+    await rif.transfer(voter.address, votingPower).then(tx => tx.wait())
+    await rif
+      .connect(voter)
+      .approve(await stRIF.getAddress(), votingPower)
+      .then(tx => tx.wait())
+    await stRIF
+      .connect(voter)
+      .depositAndDelegate(voter.address, votingPower)
+      .then(tx => tx.wait())
+  }
+
+  const deploy = async () => {
+    const contracts = await deployContracts()
+    const vanguard = (
+      await ignition.deploy(VotingVanguardsModule, {
+        parameters: {
+          VanguardNFT: {
+            maxSupply,
+            mintLimit,
+            stRifThreshold,
+            stRif: await contracts.stRIF.getAddress(),
+            governor: await contracts.governor.getAddress(),
+            proposalCount,
+            ipfsFolderCid,
+          },
+        },
+      })
+    ).VanguardNFT as unknown as VotingVanguardsRootstockCollective
+    await enfranchise(contracts.rif, contracts.stRIF)
+    return { ...contracts, vanguard }
+  }
+
+  const createProposal = async (governor: GovernorRootstockCollective) => {
+    const receipt = await governor
+      .propose([proposalTarget.address], [0n], [ZERO_BYTE], hexlify(randomBytes(32)))
+      .then(tx => tx.wait())
+    await mine(1)
+    const [event] = await governor.queryFilter(governor.filters.ProposalCreated, receipt?.blockNumber)
+    return event.args.proposalId
+  }
+
+  describe('NFT lifecycle', () => {
+    let governor: GovernorRootstockCollective
+    let vanguard: VotingVanguardsRootstockCollective
+    let stRif: StRIFToken
+
+    before(async () => {
+      const contracts = await loadFixture(deploy)
+      governor = contracts.governor
+      vanguard = contracts.vanguard
+      stRif = contracts.stRIF
+    })
+
+    describe('Upon deployment', async () => {
+      it('Contract names should be set after the deployment', async () => {
+        expect(await governor.name()).to.equal('GovernorRootstockCollective')
+        expect(await vanguard.name()).to.equal('VotingVanguardsRootstockCollective')
+      })
+      it('number of past votes to be checked should be set up', async () => {
+        expect(await vanguard.proposalCount()).to.equal(proposalCount)
+      })
+      it('StRif threshold should be set', async () => {
+        expect(await vanguard.stRifThreshold()).to.equal(stRifThreshold)
+      })
+      it('StRif address should be set in the Vanguard contract', async () => {
+        expect(await vanguard.stRif()).to.equal(await stRif.getAddress())
+      })
+      it('Mint limit should be set', async () => {
+        expect(await vanguard.mintLimit()).to.equal(mintLimit)
+      })
+      it('All tokens should be available for minting', async () => {
+        expect(await vanguard.tokensAvailable()).to.equal(maxSupply)
+      })
+      it('Number of proposal to search should be set', async () => {
```

Comment:
You have two of the same tests, look up `it('number of past votes to be checked should be set up'`